pkg/cloud/azure/actuators/machine_scope_test.go

```diff
@@ -40,7 +40,7 @@ func init() {
 }
 
 func providerSpecFromMachine(in *machineproviderv1.AzureMachineProviderSpec) (*machinev1.ProviderSpec, error) {
-	bytes, err := yaml.Marshal(in)
+	bytes, err := json.Marshal(in)
```

Comment:
why?

Reply:
Because `.Patch()` expects it to be JSON and fails if it's yaml